From: Mateus Rodrigues de Morais <mateus.morais@canonical.com>
Date: Mon, 09 Jul 2025 13:10:00 -0300
Subject: [PATCH] feat: add support for s390x and ppc64le to build script

---
 build.psm1 | 22 ++++++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/build.psm1 b/build.psm1
index ffaf7b02f9d..378bdc3f833 100644
--- a/build.psm1
+++ b/build.psm1
@@ -278,7 +278,7 @@ function Test-IsReleaseCandidate
     return $false
 }
 
-$optimizedFddRegex = 'fxdependent-(linux|win|win7|osx)-(x64|x86|arm64|arm)'
+$optimizedFddRegex = 'fxdependent-(linux|win|win7|osx)-(x64|x86|arm64|arm|s390x|ppc64le)'
 
 function Start-PSBuild {
     [CmdletBinding(DefaultParameterSetName="Default")]
@@ -320,9 +320,13 @@ function Start-PSBuild {
                      "fxdependent-noopt-linux-musl-x64",
                      "fxdependent-linux-x64",
                      "fxdependent-linux-arm64",
+                     "fxdependent-linux-ppc64le",
+                     "fxdependent-linux-s390x",
                      "fxdependent-win-desktop",
                      "linux-arm",
                      "linux-arm64",
+                     "linux-ppc64le",
+                     "linux-s390x",
                      "linux-x64",
                      "osx-arm64",
                      "osx-x64",
@@ -461,12 +465,16 @@ Fix steps:
     }
 
     # Add --self-contained due to "warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used."
-    if ($Options.Runtime -like 'fxdependent*') {
+    if ($Options.Runtime -like 'fxdependent*' -and $Options.Runtime -notmatch 's390x|ppc64le') {
         $Arguments += "--no-self-contained"
         # The UseAppHost = true property creates ".exe" for the fxdependent packages.
         # We need this in the package as Start-Job needs it.
         $Arguments += "/property:UseAppHost=true"
     }
+    elseif ($Options.Runtime -like 'fxdependent*' -and $Options.Runtime -match 's390x|ppc64le') {
+        $Arguments += "--no-self-contained"
+        $Arguments += "/property:UseAppHost=false"
+    }
     else {
         $Arguments += "--self-contained"
     }
@@ -493,9 +501,11 @@ Fix steps:
     }
 
     # Framework Dependent builds do not support ReadyToRun as it needs a specific runtime to optimize for.
+    # s390x and ppc64le builds also do not support ReadyToRun as their runtimes are Mono-based.
     # The property is set in Powershell.Common.props file.
     # We override the property through the build command line.
-    if(($Options.Runtime -like 'fxdependent*' -or $ForMinimalSize) -and $Options.Runtime -notmatch $optimizedFddRegex) {
+    if((($Options.Runtime -like 'fxdependent*' -or $ForMinimalSize) -and $Options.Runtime -notmatch $optimizedFddRegex) -or
+         $Options.Runtime -match 's390x|ppc64le') {
         $Arguments += "/property:PublishReadyToRun=false"
     }
 
@@ -990,9 +1000,13 @@ function New-PSOptions {
                      "fxdependent-noopt-linux-musl-x64",
                      "fxdependent-linux-x64",
                      "fxdependent-linux-arm64",
+                     "fxdependent-linux-ppc64le",
+                     "fxdependent-linux-s390x",
                      "fxdependent-win-desktop",
                      "linux-arm",
                      "linux-arm64",
+                     "linux-ppc64le",
+                     "linux-s390x",
                      "linux-x64",
                      "osx-arm64",
                      "osx-x64",
