From b71220779a0ec47fab9b9783cdec99a7cdb6fee6 Mon Sep 17 00:00:00 2001
From: Mateus Rodrigues de Morais <mateus.morais@canonical.com>
Date: Mon, 14 Jul 2025 17:25:23 -0400
Subject: [PATCH] feat: add support for s390x and ppc64el self-contained

---
 build.psm1 | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/build.psm1 b/build.psm1
index a610c684e..7b0c74464 100644
--- a/build.psm1
+++ b/build.psm1
@@ -323,6 +323,8 @@ function Start-PSBuild {
                      "fxdependent-win-desktop",
                      "linux-arm",
                      "linux-arm64",
+                     "ubuntu.22.04-ppc64le",
+                     "ubuntu.22.04-s390x",
                      "linux-x64",
                      "osx-arm64",
                      "osx-x64",
@@ -513,6 +515,12 @@ Fix steps:
     $Arguments += "--configuration", $Options.Configuration
     $Arguments += "--framework", $Options.Framework
 
+    # s390x and ppc64le builds do not support ReadyToRun as their runtimes are Mono-based.
+    # This property should override the <PublishReadyToRun> in the PowerShell.Common.props file.
+    if ($Options.Runtime -match 's390x|ppc64le') {
+        $Arguments += "/property:PublishReadyToRun=false"
+    }
+
     if ($Detailed.IsPresent)
     {
         $Arguments += '--verbosity', 'd'
@@ -1011,6 +1019,8 @@ function New-PSOptions {
                      "fxdependent-win-desktop",
                      "linux-arm",
                      "linux-arm64",
+                     "ubuntu.22.04-ppc64le",
+                     "ubuntu.22.04-s390x",
                      "linux-x64",
                      "osx-arm64",
                      "osx-x64",
@@ -3833,6 +3843,12 @@ function Clear-NativeDependencies
         '.*-arm64' {
             $diasymFileName = $diasymFileNamePattern -f 'arm64'
         }
+        '.*-s390x' {
+            $diasymFileName = $diasymFileNamePattern -f 's390x'
+        }
+        '.*-ppc64le' {
+            $diasymFileName = $diasymFileNamePattern -f 'ppc64le'
+        }
         'fxdependent.*' {
             Write-Verbose -Message "$($script:Options.Runtime) is a fxdependent runtime, no cleanup needed in pwsh.deps.json" -Verbose
             return
-- 
2.48.1

