From 5f505f1b103dafe0be3eab2a286eecc813d1e83a Mon Sep 17 00:00:00 2001
From: Mateus Rodrigues de Morais <mateus.morais@canonical.com>
Date: Mon, 14 Jul 2025 17:25:23 -0400
Subject: [PATCH] feat: add support for s390x and ppc64el self-contained

---
 build.psm1 | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/build.psm1 b/build.psm1
index ffaf7b02f9d..1de15affb1f 100644
--- a/build.psm1
+++ b/build.psm1
@@ -323,6 +323,8 @@ function Start-PSBuild {
                      "fxdependent-win-desktop",
                      "linux-arm",
                      "linux-arm64",
+                     "ubuntu.22.04-ppc64le",
+                     "ubuntu.22.04-s390x",
                      "linux-x64",
                      "osx-arm64",
                      "osx-x64",
@@ -493,9 +495,11 @@ Fix steps:
     }
 
     # Framework Dependent builds do not support ReadyToRun as it needs a specific runtime to optimize for.
+    # s390x and ppc64le builds also do not support ReadyToRun as their runtimes are Mono-based.
     # The property is set in Powershell.Common.props file.
     # We override the property through the build command line.
-    if(($Options.Runtime -like 'fxdependent*' -or $ForMinimalSize) -and $Options.Runtime -notmatch $optimizedFddRegex) {
+    if((($Options.Runtime -like 'fxdependent*' -or $ForMinimalSize) -and $Options.Runtime -notmatch $optimizedFddRegex) -or
+         $Options.Runtime -match 's390x|ppc64le') {
         $Arguments += "/property:PublishReadyToRun=false"
     }
 
@@ -1000,6 +1004,8 @@ function New-PSOptions {
                      "fxdependent-win-desktop",
                      "linux-arm",
                      "linux-arm64",
+                     "ubuntu.22.04-ppc64le",
+                     "ubuntu.22.04-s390x",
                      "linux-x64",
                      "osx-arm64",
                      "osx-x64",
@@ -3800,6 +3806,12 @@ function Clear-NativeDependencies
         '.*-arm64' {
             $diasymFileName = $diasymFileNamePattern -f 'arm64'
         }
+        '.*-s390x' {
+            $diasymFileName = $diasymFileNamePattern -f 's390x'
+        }
+        '.*-ppc64le' {
+            $diasymFileName = $diasymFileNamePattern -f 'ppc64le'
+        }
         'fxdependent.*' {
             Write-Verbose -Message "$($script:Options.Runtime) is a fxdependent runtime, no cleanup needed in pwsh.deps.json" -Verbose
             return
